### PR TITLE
Added 'simple' template for beginners

### PR DESCRIPTION
**Pull request by seanhess**
I've been messing around with tutorial writing for beginners: http://seanhess.github.io/2015/08/04/practical-haskell-getting-started.html

I'm also running the first Utah Haskell meetup on Thursday. I've been told by more than one person I should be using `stack new` instead of creating the files by hand, so I'm hoping to create a template that is as simple as possible for beginners.

### CODE REVIEW

**Comment by chreekat** [comment]
Never mind beginners, I'd use this all the time. :)

Question/comment: I haven't done it much, but I recently uploaded a package to Hackage with `stack upload`, and the `cabal check` process complained that I had not set a max bound on base. It might be nice to add that.

**Comment by seanhess** [comment]
Aren't there lots of things you'd have to change before it was uploadable to hackage? I'm aiming for executables that'll probably never be uploaded. Also it's kind of redundant with stack right?

**Comment by snoyberg** [comment]
Merged, and I've given you push access to be able to make changes in the future. I do agree with @chreekat about adding an upper bound on base (even if it's a meaningless `< 10` to get around the Hackage requirement of an upper bound there.

**Comment by seanhess** [comment]
Ok, I added the same base bounds as new-template. We can just update it later. Thanks guys!

**Comment by snoyberg** [comment]
Thank you for the contribution here and the ongoing documentation, I'm very
excited about it.

**Comment by bitemyapp** [comment]
@seanhess this looks great. Would it be useful if there's a library variation of the same?

At this rate, I'm going to have to rewrite the parts of our book that write Cabal files manually :smile:

**Comment by seanhess** [comment]
Yes I think so! I haven't made enough libraries to be sure what it
requires, but I think it's a good idea.

**Comment by bitemyapp** [comment]
@seanhess I'll take a whack soon.

**Comment by tebello-thejane** [comment]
Hmm... Why is this defaulting to the ``BSD3`` licence? 

Certainly, ``cabal new``'s workflow in this regard leaves a lot to be desired, but if you're going to choose a default, is there a special reason why you defaulted to ``BSD3``?

**Comment by chrisdone** [comment]
That's what the Haskell community likes, most packages on Hackage are BSD3, including GHC.

**Comment by tebello-thejane** [comment]
Is there a deeper reason than that? 

Not that I am totally confused by open source licences. I know that the ``GPL`` causes a lot of grief, and as a safe bet I just release everything in the public domain. So please ELI5. :-)

**Comment by chreekat** [comment]
Curiously, the public domain is actually not a safe option, since some
jurisdictions have no such concept, effectively making your work unlicensed.

One reason the BSD was chosen is that GHC is known to inline significant
chunks of code from underlying libraries. Thus, any piece of the ecosystem
has to grant just as many rights to its users as the most lenient[1] of its
dependencies. Since a lot of Hackage code is used to make proprietary
software, there was probably pressure to keep available the option of
restricting users' rights — so the entire ecosystem settled on a license
that makes it possible to do so.

[1]: Speaking from the point of view of the users here. Obviously the
reverse is true (the BSD is more lenient) from the *developers'* viewpoint.
The two communities are not always aligned. :)

**Comment by bitemyapp** [comment]
Thank you @chreekat - I wish more people understood public domain doesn't work for a lot of countries.

@tebello-thejane I use BSD/MIT/Apache by default in my projects because I am mostly writing things I'd like to be able to use in my day-to-day work. I am not often writing stuff that has "end users", just fellow developers who won't be able to use it if it's GPL'd.

**Comment by tebello-thejane** [comment]
Makes sense.

I really just want people to [do whatever the fuck they want](http://www.wtfpl.net/txt/copying/). I've researched the BSD3 a little bit and it looks like that's the best license to allow just that.
